src/sentry/tasks/embeddings_grouping/utils.py

```diff
@@ -126,46 +126,48 @@ def initialize_backfill(
     else:
         last_processed_project_index_ret = last_processed_project_index
 
-    return project, last_processed_group_index_ret, last_processed_project_index_ret
+    return project, last_processed_group_id_ret, last_processed_project_index_ret
 
 
 @sentry_sdk.tracing.trace
-def get_current_batch_groups_from_postgres(project, last_processed_group_index, batch_size):
-    groups_to_backfill_query = (
+def get_current_batch_groups_from_postgres(project, last_processed_group_id, batch_size):
+    group_id_filter = Q()
+    if last_processed_group_id is not None:
+        group_id_filter = Q(id__lt=last_processed_group_id)
+
+    groups_to_backfill_query = RangeQuerySetWrapper(
         Group.objects.filter(
+            group_id_filter,
             project_id=project.id,
             type=ErrorGroupType.type_id,
             times_seen__gt=1,
             last_seen__gt=(datetime.now(UTC) - timedelta(days=90)),
-        )
-        .exclude(status__in=[GroupStatus.PENDING_DELETION, GroupStatus.DELETION_IN_PROGRESS])
-        .values_list("id", "data")
-        .order_by("-times_seen", "id")
+        ).exclude(status__in=[GroupStatus.PENDING_DELETION, GroupStatus.DELETION_IN_PROGRESS]),
+        step=-1000,
+        limit=batch_size,
+    )
+    groups_to_backfill_batch = [(group.id, group.data) for group in groups_to_backfill_query]
```

Comment:
Since I think you're only using `group.id` and `group.data` you can use `values_list` to bring down only the fields you need.